### PR TITLE
Nsng 97 video in story

### PR DESCRIPTION
**Pull request by igor-kamil**
Snímka obrazovky 2022-12-06 o 3 16 07

Snímka obrazovky 2022-12-06 o 3 16 15

### CODE REVIEW

File: resources/js/components/StoryVideoLightbox.vue [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        </button>
+    </a>
+    <div class="fixed inset-0 z-50 flex justify-center items-center" v-if="visible" @click="visible = false">
+        <div class="bg-black opacity-70 absolute inset-0 cursor-zoom-out" @click="visible = false" />
+        <div class="relative max-h-full w-full md:max-w-lg mx-4" @click.stop="visible = false">
+            <div :style="'aspect-ratio:' + story.video_aspect_ratio.width + ' / ' + story.video_aspect_ratio.height">
+                <iframe class="rounded-xl w-full h-full" :src="story.video_embed" frameborder="0" allow="autoplay;"
+                    allowfullscreen></iframe>
+            </div>
+            <div class="absolute cursor-pointer p-1 top-0 right-0 bg-white rounded-tr-xl" @click.stop="visible = false">
```
Comment (rastislav-chynoransky) [inline resources/js/components/StoryVideoLightbox.vue:17]: Change to `button` to make it accessible

File: resources/js/components/StoryVideoLightbox.vue [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        <ResponsiveImage class="rounded-xl w-full object-cover cursor-pointer" :image="story.video_thumbnail" />
+        <button
+            class="bg-green rounded-xl px-3 py-2 bottom-3 right-3 absolute flex items-center appearance-none text-[16px] text-black font-medium">
+            <SvgPlay class="mr-2" />
+            {{ $t('Play video') }} ({{ story.video_duration }})
+        </button>
+    </a>
+    <div class="fixed inset-0 z-50 flex justify-center items-center" v-if="visible" @click="visible = false">
+        <div class="bg-black opacity-70 absolute inset-0 cursor-zoom-out" @click="visible = false" />
+        <div class="relative max-h-full w-full md:max-w-lg mx-4" @click.stop="visible = false">
```
Comment (rastislav-chynoransky) [inline resources/js/components/StoryVideoLightbox.vue:12]: `max-h-full` here is not really working. Also why the click callback here?

File: resources/js/components/StoryVideoLightbox.vue [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<template>
+    <a href="#" @click.prevent="visible = true">
+        <ResponsiveImage class="rounded-xl w-full object-cover cursor-pointer" :image="story.video_thumbnail" />
+        <button
+            class="bg-green rounded-xl px-3 py-2 bottom-3 right-3 absolute flex items-center appearance-none text-[16px] text-black font-medium">
+            <SvgPlay class="mr-2" />
+            {{ $t('Play video') }} ({{ story.video_duration }})
+        </button>
+    </a>
+    <div class="fixed inset-0 z-50 flex justify-center items-center" v-if="visible" @click="visible = false">
```
Comment (rastislav-chynoransky) [inline resources/js/components/StoryVideoLightbox.vue:10]: This flex container has only two direct children and the first one is absolute positioned, I'd move the container deeper.

File: resources/js/components/InteractionStory.vue [hunk 4]
```diff
@@ -8,26 +8,22 @@
             <ResponsiveImageWithPlaceholder class="rounded-xl w-full" :image="image" />
         </div>
 
-        <div class="my-4" v-if="story.video_thumbnail">
-            <ResponsiveImage class="rounded-xl w-full" :image="story.video_thumbnail" />
+        <div class="my-4 relative" v-if="story.video_embed">
```
Comment (rastislav-chynoransky) [inline resources/js/components/InteractionStory.vue:11]: I'd move the `relative` class inside `StoryVideoLightbox` as that's its responsibility.

File: resources/js/components/StoryVideoLightbox.vue [hunk 5]
```diff
@@ -0,0 +1,32 @@
+<template>
+    <a href="#" @click.prevent="visible = true">
```
[the remark this answers is not in the document]
Comment (rastislav-chynoransky) [reply resources/js/components/StoryVideoLightbox.vue:2]: Also this link has to be disabled when the `InteractionStory` has `active === true` prop.

**Review by rastislav-chynoransky — CHANGES_REQUESTED** [review]

File: resources/js/components/StoryVideoLightbox.vue [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        <ResponsiveImage class="rounded-xl w-full object-cover cursor-pointer" :image="story.video_thumbnail" />
+        <button
+            class="bg-green rounded-xl px-3 py-2 bottom-3 right-3 absolute flex items-center appearance-none text-[16px] text-black font-medium">
```
[the remark this answers is not in the document]
Comment (igor-kamil) [reply resources/js/components/StoryVideoLightbox.vue:5]: definitely! I was just thinking about that, when I was writing this

File: resources/js/components/ItemImageLightbox.vue [hunk 7]
```diff
@@ -1,43 +1,27 @@
 <template>
-    <a href="#" @click.prevent="visible = true">
+    <div role="link" class="relative" @click.prevent="visible = true">
```
Comment (rastislav-chynoransky) [inline resources/js/components/ItemImageLightbox.vue:2]:
```suggestion
    <div role="link" class="relative" @click="visible = true">
```

File: resources/js/components/StoryVideoLightbox.vue [hunk 8]
```diff
@@ -0,0 +1,32 @@
+<template>
+    <div role="link" class="relative" @click.prevent="active && (visible = true)">
+        <ResponsiveImage class="rounded-xl w-full object-cover cursor-pointer" :image="story.video_thumbnail" />
+        <button
+            class="bg-green rounded-xl px-3 py-2 bottom-3 right-3 absolute flex items-center appearance-none text-[16px] text-black font-medium">
+            <SvgPlay class="mr-2" />
+            {{ $t('Play video') }} ({{ story.video_duration }})
+        </button>
+    </div>
+    <div class="fixed inset-0 z-50 flex justify-center items-center" v-if="visible" @click="visible = false">
```
Comment (rastislav-chynoransky) [inline resources/js/components/StoryVideoLightbox.vue:10]:
```suggestion
    <div class="fixed inset-0 z-50 flex justify-center items-center" v-if="visible">
```

Comment (igor-kamil) [reply resources/js/components/StoryVideoLightbox.vue:10]: I think it's good to close lightbox also by clicking "outside" (also the same behaviour is used in ItemImageLightbox)

File: resources/js/components/StoryVideoLightbox.vue [hunk 9]
```diff
@@ -0,0 +1,32 @@
+<template>
+    <div role="link" class="relative" @click="active && (visible = true)">
```
[the remark this answers is not in the document]
Comment (igor-kamil) [reply resources/js/components/StoryVideoLightbox.vue:2]: I was thinking about it. First I wanted to make it possible to open video anytime. 
Anyway - I think this is a broader topic. Keep it consistent. If story has inactive byttons - keep it all inactive. 
Sooner or later we shoul make it possible to return back to stroy imho.

[the remark this answers is not in the document]
Comment (rastislav-chynoransky) [reply resources/js/components/StoryVideoLightbox.vue:2]: Without the lowered opacity it will appear broken, but I can do it in the recoloring branch.

**Review by rastislav-chynoransky — CHANGES_REQUESTED** [review]
Please resolve conflicts and it's good to go.

**Pull request merged**